sdks/python/apache_beam/transforms/core.py

```diff
@@ -2779,14 +2779,20 @@ def StripNonce(nonce_key_value):
 
     cold, hot = pcoll | ParDo(SplitHotCold()).with_outputs('hot', main='cold')
     cold.element_type = typehints.Any  # No multi-output type hints.
+    precombined_hot = hot
+    if pcoll.windowing.accumulation_mode == AccumulationMode.ACCUMULATING:
+      # Avoid double-counting that may happen with stacked accumulating mode.
+      precombined_hot |= 'WindowIntoDiscarding' >> WindowInto(
+          pcoll.windowing, accumulation_mode=AccumulationMode.DISCARDING)
+
     precombined_hot = (
-        hot
-        # Avoid double counting that may happen with stacked accumulating mode.
-        | 'WindowIntoDiscarding' >> WindowInto(
```

Comment:
@robertwb Regarding your suggestion offline to use _IdentityWindowFn. Did you mean to use instead:
```
 precombined_hot = (
            hot
            | 'WindowIntoIdentityAndDiscarding' >> WindowInto(
              _IdentityWindowFn(pcoll.windowing.windowfn.get_window_coder()), accumulation_mode=AccumulationMode.DISCARDING)
            | CombinePerKey(PreCombineFn())
            | Map(StripNonce)
            | 'WindowIntoOriginal' >> WindowInto(pcoll.windowing))


    return ((cold, precombined_hot)
            | Flatten()
            | CombinePerKey(PostCombineFn()))

```
?

Reply:
I tried that,  seems that using that transform makes tests fail, i can dig deeper to undestand the failure mode but wanted to check if I understood your intent correctly

Reply:
another suggestion to modify accumulation_mode directly via `hot.windowing.accumulation_mode = AccumulationMode.DISCARDING` also doesn't work, maybe because this is assignment is then reverted while still at pipeline construction time, also it has side-effect on other pcolls (e.g. cold), and hot.windowing is not assignable directly

Reply:
I think current PR is the best fix I have so far, alternative is to skip `WindowIntoDiscarding` explicitly if `SlidingWindows` are detected.

Reply:
TODO: try to keep the same trigger, and/or other properties on the windowing mode.

Reply:
Updated per offline conversation.